packages/core/src/media/metrics-new.ts

```diff
@@ -0,0 +1,578 @@
+import { warn } from "../utils";
+import { MediaControllerStore } from "./controller";
+import { getMetricsReportingPOSTUrl } from "./metrics-utils";
+import { MimeType } from "./mime";
+
+export type PlaybackEvent =
+  | {
+      /** The event type. */
+      type: "heartbeat";
+      /** The timestamp of the event, in milliseconds. */
+      timestamp: number;
+      /** The number of *total* errors that have occurred. */
+      errors: number;
+      /** The state of autoplay of the video. */
+      autoplay_status: "autoplay" | "none";
+
+      /** The time from the metrics were added to play, in milliseconds. */
+      mount_to_play_ms?: number;
+      /** The time from the metrics were added to the first frame, in milliseconds. */
+      mount_to_first_frame_ms?: number;
+      /** The time from the first play event to the first frame, in milliseconds. Also called TTFF. */
+      play_to_first_frame_ms?: number;
+
+      /** The duration of the video, in milliseconds. */
+      duration_ms?: number;
+      /** The offset of the live video head compared to the server time, in milliseconds. */
+      offset_ms?: number;
+
+      /** The *total* number of times the current playback has stalled. */
+      stalled_count: number;
+      /** The *total* number of times the current playback has waited. */
+      waiting_count: number;
+
+      /** The *total* time the playback has spent stalled, in ms. */
+      time_stalled_ms: number;
+      /** The *total* time the playback has spent playing, in ms. */
+      time_playing_ms: number;
+      /** The *total* time the playback has spent waiting, in ms. */
+      time_waiting_ms: number;
+
+      /** The height of the player element, in px. */
+      player_height_px?: number;
+      /** The width of the player element, in px. */
+      player_width_px?: number;
+      /** The height of the source video, in px. */
+      video_height_px?: number;
+      /** The width of the source video, in px. */
+      video_width_px?: number;
+      /** The height of the window, in px. */
+      window_height_px?: number;
+      /** The width of the window, in px. */
+      window_width_px?: number;
+    }
+  | {
+      type: "error";
+      timestamp: number;
+      error_message: string;
+    };
+
+export type SessionData = {
+  session_id: string;
+  playback_id: string;
+  protocol?: MimeType;
+  page_url: string;
+  source_url: string;
+  player: `${"audio" | "hls" | "video" | "webrtc" | "unknown"}-${string}`;
+  user_agent?: string;
+  uid?: string;
+  events: PlaybackEvent[];
+};
+
+const getCurrentTimestamp = () => Math.floor(Date.now() / 1000);
```

Comment:
Ideally this would be milliseconds. Is that what this is or are you rounding to seconds?

Reply:
Yeah we can do ms. In the spec in the notion doc, it's seconds. @leszko may have insight into that

Reply:
If it's no additional work from you Chase, then let's just send ms.